pmlc/dialect/affinex/transforms/dead_memref_elimination.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "pmlc/dialect/affinex/transforms/pass_detail.h"
+
+using namespace mlir; // NOLINT
+
+namespace pmlc::dialect::affinex {
+
+struct AffinexDeadMemRefElimination
+    : public AffinexDeadMemRefEliminationBase<AffinexDeadMemRefElimination> {
+
+  void runOnFunction() override {
+    llvm::SmallVector<Operation *, 8> opsToErase;
+    getFunction().walk([&](AllocOp alloc) {
+      auto memref = alloc.getResult();
+      for (auto user : memref.getUsers()) {
```

Comment:
This should be `auto *` or even better: `Operation*`

Reply:
Done